File: WebApi/WebApi/Models/Purchase.cs
namespace WebApi.Models
{
    public enum BasketStatus
    {
        Draft = 0,
        Confirmed = 1
    }

    public class Purchase
    {
        public int Id { get; set; }
        public DateTime Date { get; set; } = DateTime.Now;

        public int GiftId { get; set; }
        public Gift? Gift { get; set; }

        public int UserId { get; set; }
        public User? User { get; set; }

        public static decimal TotalSum { get; set; } = 0;

        public BasketStatus basketStatus { get; set; } = BasketStatus.Draft;
    }
}

File: Angular/Angular/src/Model/Purchase.ts
import { Gift } from "./Gift";
import { User } from "./User";

export class Purchase {
    public id: number=0;
    public date: Date = new Date();
    public giftId: number=0;
    public gift: Gift=new Gift();
    public userId: number=0;
    public user: User=new User();
    public static totalSum: number = 0;
    public isRandom: boolean = false;
    
}